app/serializers/user_auth_token_serializer.rb

```diff
@@ -0,0 +1,88 @@
+class UserAuthTokenSerializer < ApplicationSerializer
+  attributes :id,
+             :action,
+             :client_ip,
+             :created_at,
+             :seen_at,
+             :os,
+             :device_name,
+             :icon
+
+  def action
+    case object.action
+    when 'generate'
+      I18n.t('log_in')
+    when 'destroy'
+      I18n.t('unsubscribe.log_out')
+    else
+      I18n.t('staff_action_logs.unknown')
+    end
+  end
+
+  def include_action?
+    object.has_attribute?(:action)
+  end
+
+  def client_ip
+    object.client_ip.to_s
+  end
+
+  def include_seen_at?
+    object.has_attribute?(:seen_at)
```

Comment:
@nbianca Are we passing objects that are not instances of `UserAuthToken` to the serializer? I'll like to clarify why we need to check if the object has the `seen_at` attribute.

Reply:
I think @nbianca meant `object.seen_at.present?` ;)

Reply:
Nope. This is right. I am using the serializer for `UserAuthTokenLogs`, which does not have a `seen_at` column. :)

Reply:
@nbianca Usually we want each serializer to have a single reponsibility. That means `UserAuthTokenSerializer` should only be expected to receive `UserAuthToken` objects for serialization. The trouble when a serializer is made to support different kind of objects is that we end up having to maintain conditionals which in turn complicates testing. I understand that the purposes of supporting different objects here is meant to avoid repeating code which is great :+1:  However, I think extracting the common logic into a module will have been easier to reason about and still allow us to have `UserAuthTokenSerializer` and `UserAuthTokenLogSerializer`. Do have a look through the changes in https://github.com/discourse/discourse/commit/7256b3bf689aa156086a397ca5264188a3d912ed to see how you can easily include class and instance methods easily via a module next time.